### DISCUSSION TITLE
Extend character limit for entity card values to 16 chars (or more)

### DISCUSSION THREAD

**Discussion by neuhausf** [Blackymas/NSPanel_HA_Blueprint#307] [Polls]
**Please vote!**

For the representation of entity values there are the new entity cards.
The **_values_** can currently have a maximum of 10 characters. This is sufficient for most use cases. BUT:

To be able to display longer/combined values, I propose the following change:
Extend the maximum number of characters to 16 characters. Aligning the values on the right - this makes the best use of the available space and the display is also appealing (used this way in HA in the weather view for example, see screenshot).

Pros:
- Combined display of temperature / humidity (see screenshot, yes this is my usecase 😉 )
- Wind speed and direction (see screenshot, `"11.5 km/h (NNO)"`)
- Status as strings (location of persons `"Grandparents"`, date with day `"Montag 14.12.2023"`)

Cons:
- no? labels should still have enough place

(note that in the following photo, the values are centered - I propose to ***right align them***)
<img width="469" alt="image" src="https://user-images.githubusercontent.com/7963239/216975495-38aaa985-3e76-42a4-aa62-2dbc0c2c8314.png">

(current version)
<img width="694" alt="image" src="https://user-images.githubusercontent.com/7963239/216975873-0ba3c32a-72a4-402d-8065-a916693911c6.png">

<img width="397" alt="image" src="https://user-images.githubusercontent.com/7963239/216974121-be1e17b9-b88c-418a-9221-c86cb4122d73.png">

See also this issue for reference: https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/300

**Comment by Blackymas** [comment]
thanks for the poll ;)

short info that it is clear for all.
Increasing the characters for Value decreases the characters for the label

**Reply by Statusbit** [reply]
The missing dot at the date on the main screen ... same reason?

**Reply by Blackymas** [reply]
Do you have a screen

**Reply by neuhausf** [reply]
I guess he means the screenshot in the README

![](https://user-images.githubusercontent.com/41958506/212768891-9610d800-43c5-454f-a8b6-6b7c43677b5d.png)

**Reply by Blackymas** [reply]
No that has nothing to do with it. The point is simply not set in the code.

Makes no sense Friday is shorter than Wednesday ;)

timestamp_custom('%a %d.%m'

would then look like this

timestamp_custom('%a %d.%m.'


do you need that? so I know that with some also so without the point - wasted space  ;)

**Reply by Statusbit** [reply]
For me there belongs a dot - but I can live without.

**Reply by neuhausf** [reply]
Same for me. Just add the dot 😉

**Comment by Statusbit** [comment]
![Screenshot_20230207-204716](https://user-images.githubusercontent.com/85311570/217350046-1d7df93e-73f3-4380-8da4-a4495ec4f3d3.png)
Yes - but not only on this Screenshot ... "Dienstag, 7.2" why not "Dienstag, 7.2."?

**Comment by edwardtfn** [comment]
One option is to adjust the font size based on the number of chars to be shown. Each char takes a different size, but I think we can get to something that is good enough.
That field is currently using a font which is size 26, but we have available sizes 22, 18 and 16 also.

**Reply by neuhausf** [reply]
That would be really great! Right now I‘m missing just 2 chars to display temp and humidity in combination.

**Reply by edwardtfn** [reply]
Do you mind creating this as an enhancement request on the Issues area? For some reason I'm not able to directly convert this discussion to an Issue, where we are tracking Enhacement requests.
This is the link:
- https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/new?assignees=&labels=Enhancement&projects=&template=enhancement.yml&title=%60Enhancement%60+

**Reply by neuhausf** [reply]
Done: https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/1728

**Comment by edwardtfn** [comment]
I'm looking for thoughts on the enhancement request:
- #1728
